String comparision should not be done with '=='

Sounds like a students job who missed a class or
drank too much beer ;-)

Signed-off-by: Remy Bohmer <linux@bohmer.net>
---
 arch/arm/plat-mxc/usb_common.c |   14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

Index: linux-2.6.26/arch/arm/plat-mxc/usb_common.c
===================================================================
--- linux-2.6.26.orig/arch/arm/plat-mxc/usb_common.c	2009-04-23 21:10:02.000000000 +0200
+++ linux-2.6.26/arch/arm/plat-mxc/usb_common.c	2009-04-23 21:13:45.000000000 +0200
@@ -480,14 +480,14 @@ int fsl_usb_host_init(struct platform_de
 	} else if (xops->xcvr_type == PORTSC_PTS_ULPI) {
 		if (cpu_is_mx51()) {
 #ifdef CONFIG_USB_EHCI_ARC_H1
-		if (pdata->name == "Host 1") {
+		if (!strcmp(pdata->name, "Host 1")) {
 			usbh1_set_ulpi_xcvr();
 			if (cpu_is_mx51())
 			gpio_usbh1_setback_stp();
 			}
 #endif
 #ifdef CONFIG_USB_EHCI_ARC_H2
-		if (pdata->name == "Host 2") {
+		if (!strcmp(pdata->name, "Host 2")) {
 			usbh2_set_ulpi_xcvr();
 			if (cpu_is_mx51())
 				gpio_usbh2_setback_stp();
@@ -497,7 +497,7 @@ int fsl_usb_host_init(struct platform_de
 			usbh2_set_ulpi_xcvr();
 	}
 
-	if (pdata->name == "Host 2")
+	if (!strcmp(pdata->name, "Host 2"))
 		/* disable remote wakeup irq */
 		USBCTRL &= ~UCTRL_H2WIE;
 
@@ -834,9 +834,9 @@ int usb_wakeup_irq(struct device *wkup_d
 	int wakeup_req = 0;
 	struct fsl_usb2_platform_data *pdata = wkup_dev->platform_data;
 
-	if (pdata->name == "Host 2")
+	if (!strcmp(pdata->name, "Host 2"))
 		wakeup_req = USBCTRL & UCTRL_H2WIR;
-	else if (pdata->name == "DR")
+	else if (!strcmp(pdata->name, "DR"))
 		wakeup_req = USBCTRL & UCTRL_OWIR;
 
 	return wakeup_req;
@@ -847,12 +847,12 @@ void usb_wakeup_set(struct device *wkup_
 {
 	struct fsl_usb2_platform_data *pdata = wkup_dev->platform_data;
 
-	if (pdata->name == "Host 2") {
+	if (!strcmp(pdata->name, "Host 2")) {
 		if (para)
 			USBCTRL |= UCTRL_H2WIE;
 		else
 			USBCTRL &= ~UCTRL_H2WIE;
-	} else if (pdata->name == "DR") {
+	} else if (!strcmp(pdata->name, "DR")) {
 		if (para)
 			USBCTRL |= UCTRL_OWIE;
 		else
